Removing BlendingInfo struct and allowing custom blending modes

Doddler opened the pull request:
The two structs `BlendingInfo` class and `BlendingParameters` share a similar purpose and contain nearly identical data, with the exception that `BlendingInfo` breaks down the `BlendingMode` into individual blending factor settings. This abstraction makes it so that setting custom blending modes was not possible however, and if you add that back into the `BlendingParameters` struct, then the two structs contain the same data and are largely redundant.

To solve that, I merged the data from `BlendingInfo` into `BlendingParameters`, and updated the usage of the struct. A new `BlendingMode.Custom` was added as well, which allows you to provide a custom set of blending factors to use rather than be limited to the default types.

I've also updated the Blending visual test to allow selecting any of the custom blending modes.

peppy commented:
Thanks for this contribution!

On a quick look, it seems that we may want to consider not even having `BlendingFactors`, and just combining that into `BlendingParameters` itself.

We could then remove the "Custom" blend mode (along with the others) and expose the default blend modes as  static get-properties inside `BlendingParameters`, something like:

![image](https://user-images.githubusercontent.com/191335/63208214-4a30ef00-c10c-11e9-80d1-290a2fdde032.png)

See `FontAwesome.cs` for an example of this pattern.

It would mean this becomes a breaking change, but likely lead to clearer usage. We could specify the default blend types using the new statics.

Doddler commented:
Thanks Peppy, based on your recommendation I've modified `BlendingParameters` to include the 4 factors using the new enum `BlendingType` rather than the OpenTK values. Using the new enum allows us to have uninitialized `BlendingParameters` automatically default to Inherit on all 4 fields. Get properties are also added for each of the commonly used blending types (None/Inherit/Mixture/Additive). 

To translate `BlendingTypes` to src/dest blending factors used by OpenTK, I added get properties which perform the conversion. Lastly, I added two helper methods. `CopyFromParent` which copies all properties marked as inherit from a parent `BlendingParameters`, and `ApplyDefaultToInherited` which will fill any field marked as inherited with default settings. This last one is important because root objects using a default `BlendingParameters` with all values set to Inherit is still expected to provide a valid blending type that can be inherited by it's children.

Unfortunately, as you said this set of changes makes it a breaking change. In most usage scenarios I found that this should simply require changing (as an example) `BlendingMode.Additive` to `BlendingParameters.Additive` and so forth, but there may be more complicated scenarios.

peppy commented:
Made a few code quality tidy-ups and added back a compatibility layer to allow compatibility with old code. Looks good otherwise 👍
